public/logo-finder.js

```diff
@@ -0,0 +1,57 @@
+export class logoFinder {
```

Comment:
👋 I love classes, this way of implementing the feature might be much better and much more efficient than the initial implementation, but could you please reimplement the feature while following the initial design ? Thanks!